### ISSUE TITLE
Adaptative value scales on LineChart

### ISSUE THREAD

**Issue by alexisbcc** [britecharts/britecharts#866]
Hi, I'm creating a line chart with values between 20 and 80, and I would like to limit the y axis to this values.

The current graph shows the y axis scale from 0 to 80.

![linechart](https://user-images.githubusercontent.com/38666191/97067329-e14fcd00-1581-11eb-93da-8b17b41007c4.PNG)

Is possible to set yaxis limits from 20 to 80?

**Comment by Golodhros** [comment]
Hi @alexisbcc, thanks for your question!

It is not currently possible to set the initial Y axis to 0. We support negative values, but when the values are positive, we default the y axis to 0, as it is the usual best practice on data visualization.

What is your use case? In theory, it wouldn't be hard to add an option to make it possible, but this was never asked before.

**Comment by alexisbcc** [comment]
Hi @Golodhros.

I'm currently making a monitoring app, it's  very important to see the variation in the values.

For example when the values have a max value of 80 and a min of 75, with the current y axis limits [0, 80] it is very difficult to appreciate the variation of those values, but if this limits could be set to [75, 80] would help a lot.

A: y axis limits by default are set to [0, 80].
B: y axis limits set to [75, 80].
![Untitled Diagram(2)](https://user-images.githubusercontent.com/38666191/97098904-4d990200-1650-11eb-9e8b-697dbb1e40c8.png)

Thank you for your quick response.

**Comment by Golodhros** [comment]
I see, we could add this.

I guess it would be an boolean option, maybe 'hasAdaptativeValueScale`, `hasMinimumValueScale` or similar, making the scale to start on the minimum value when it is true.

**Comment by alexisbcc** [comment]
Yes that could help to solve the problem.

I'm not sure if an option to set the limits manually would be needed, but for this use case _hasAdaptativeValueScale, hasMinimumValueScale`_ will do the trick.

I'm not sure if I could help because I don't have that much experience but if I could would be great.

**Comment by alexisbcc** [comment]
Hello, I tried making some changes to implement this.

Added hasMinimumValueScale with default value of false.

```javascript
let hasMinimumValueScale = false;
```
In buildYAxisScale():
```javascript
let yScaleBottomValue = minY < 0 || hasMinimumValueScale ? minY : 0;
```

And the public function:
```javascript
exports.hasMinimumValueScale = function (_x) {
    if (!arguments.length) {
        return hasMinimumValueScale;
    }
    hasMinimumValueScale = _x;

    return this;
};
```
This approach is similar to isAnimated parameter, it work as intended, but I'm not sure if it's the best way to implement.

**Comment by Golodhros** [comment]
Hi @alexisbcc, sorry for the delay, I was on leave lately!

That looks good to me! Thanks for following a well-established pattern in the code. I'm glad it was simple enough!

Just send this with some tests and we'll get it in!

Thanks!

**Comment by alexisbcc** [comment]
Hi @Golodhros I'll make some test and post here just to check if I'm doing it right.

**Comment by alexisbcc** [comment]
Hello @Golodhros, I'm a bit lost here, how should I create the new feature branch? I tried creating a PR from the current branch `:feat-adaptive-value-scales-linechart` to `release/3.0.0-rc6` but it has some commits.

**Comment by Golodhros** [comment]
> Hello @Golodhros, I'm a bit lost here, how should I create the new feature branch? I tried creating a PR from the current branch `:feat-adaptive-value-scales-linechart` to `release/3.0.0-rc6` but it has some commits.

Hi @alexisbcc,

To do this, I would recommend you to checkout 'release/3.0.0-rc6' in your machine, create a new branch from it, and using the code changes on this PR, copy/paste them on the new branch. Then you can push that branch up and create a PR against 'release/3.0.0-rc6'.

I think this will be the less problematic way!

**Comment by alexisbcc** [comment]
Hello @Golodhros I created a PR against 'release/3.0.0-rc6', please let me know any comments and suggestions.

**Comment by Golodhros** [comment]
Fixed on #971
